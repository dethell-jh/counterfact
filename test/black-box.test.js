/* eslint-disable jest/no-hooks */
import { exec } from "node:child_process";

// eslint-disable-next-line @typescript-eslint/no-shadow, no-shadow
import fetch from "node-fetch";

describe("black box test", () => {
  // eslint-disable-next-line init-declarations
  let counterfactProcess;

  beforeAll(async () => {
    // eslint-disable-next-line sonar/no-os-command-from-path
    counterfactProcess = exec("npx . ./openapi-example.yaml out");

    // eslint-disable-next-line promise/avoid-new
    await new Promise((resolve) => {
      counterfactProcess.stdout.on("data", resolve);
    });

    // wait a few seconds to make sure the server is up
    // eslint-disable-next-line promise/avoid-new
    await new Promise((resolve) => {
      setTimeout(resolve, 5000);
    });
  }, 60_000);

  afterAll(() => {
    counterfactProcess.stdout.destroy();
    counterfactProcess.stderr.destroy();
    counterfactProcess.stdin.destroy();
    counterfactProcess.kill();
    counterfactProcess.unref();
  });

  it("responds to a GET request", async () => {
    const response = await fetch("http://localhost:3100/hello/world");

    expect(response.status).toBe(200);
  });
});
