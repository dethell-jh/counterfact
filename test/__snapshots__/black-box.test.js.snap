// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`black box test compiles kitty.ts 1`] = `
Object {
  "data": Array [
    101,
    120,
    112,
    111,
    114,
    116,
    32,
    118,
    97,
    114,
    32,
    71,
    69,
    84,
    32,
    61,
    32,
    102,
    117,
    110,
    99,
    116,
    105,
    111,
    110,
    32,
    40,
    36,
    41,
    32,
    123,
    10,
    32,
    32,
    32,
    32,
    114,
    101,
    116,
    117,
    114,
    110,
    32,
    36,
    46,
    114,
    101,
    115,
    112,
    111,
    110,
    115,
    101,
    91,
    50,
    48,
    48,
    93,
    46,
    114,
    97,
    110,
    100,
    111,
    109,
    40,
    41,
    59,
    10,
    125,
    59,
    10,
  ],
  "type": "Buffer",
}
`;

exports[`black box test creates a file for the /hello/kitty path 1`] = `
Object {
  "data": Array [
    105,
    109,
    112,
    111,
    114,
    116,
    32,
    116,
    121,
    112,
    101,
    32,
    123,
    32,
    72,
    84,
    84,
    80,
    95,
    71,
    69,
    84,
    32,
    125,
    32,
    102,
    114,
    111,
    109,
    32,
    34,
    46,
    46,
    47,
    46,
    46,
    47,
    112,
    97,
    116,
    104,
    45,
    116,
    121,
    112,
    101,
    115,
    47,
    104,
    101,
    108,
    108,
    111,
    47,
    107,
    105,
    116,
    116,
    121,
    46,
    116,
    121,
    112,
    101,
    115,
    46,
    106,
    115,
    34,
    59,
    10,
    10,
    101,
    120,
    112,
    111,
    114,
    116,
    32,
    99,
    111,
    110,
    115,
    116,
    32,
    71,
    69,
    84,
    58,
    32,
    72,
    84,
    84,
    80,
    95,
    71,
    69,
    84,
    32,
    61,
    32,
    40,
    36,
    41,
    32,
    61,
    62,
    32,
    123,
    10,
    32,
    32,
    114,
    101,
    116,
    117,
    114,
    110,
    32,
    36,
    46,
    114,
    101,
    115,
    112,
    111,
    110,
    115,
    101,
    91,
    50,
    48,
    48,
    93,
    46,
    114,
    97,
    110,
    100,
    111,
    109,
    40,
    41,
    59,
    10,
    125,
    59,
    10,
  ],
  "type": "Buffer",
}
`;

exports[`black box test responds to a GET request 1`] = `"<img src=\\"https://upload.wikimedia.org/wikipedia/en/0/05/Hello_kitty_character_portrait.png\\">"`;
